UnitystationLauncher/ViewModels/SecurityPanelViewModel.cs

```diff
@@ -0,0 +1,538 @@
+            {
+                DeleteContentsOfDirectory(processingDirectory);
+                return;
+            }
+
+            var pluginsFolder = dataPath.CreateSubdirectory("Plugins");
+            pluginsFolder.Delete(true);
+
+            pluginsFolder = dataPath.CreateSubdirectory("Plugins");
+
+            DirectoryInfo goodPluginsCopy = new DirectoryInfo(Path.Combine(System.AppDomain.CurrentDomain.BaseDirectory,
+                "GoodFiles", "VDev", "Plugins"));
+
+            //TODO OS Switch?
+            CopyFilesRecursively(goodPluginsCopy.ToString(), pluginsFolder.ToString());
+
+
+            var monoBleedingEdge = processingDirectory.CreateSubdirectory("MonoBleedingEdge");
+
+            monoBleedingEdge.Delete(true);
+
+
+            var goodRoot = new DirectoryInfo(Path.Combine(System.AppDomain.CurrentDomain.BaseDirectory,
+                "GoodFiles", "VDev", "Root"));
+
+            try
+            {
+                CopyFilesRecursively(goodRoot.ToString(), processingDirectory.ToString());
+            }
+            catch (Exception e)
+            {
+                errors.Invoke("e > " + e);
+                DeleteContentsOfDirectory(processingDirectory);
+                return;
+            }
+
+
+            var exeRename = processingDirectory.GetFiles()
+                .FirstOrDefault(x => x.Extension == ".exe" && x.Name != "UnityCrashHandler64.exe"); //TODO OS
+
+            if (exeRename == null || exeRename.Directory == null)
+            {
+                errors.Invoke("no Executable found ");
+                DeleteContentsOfDirectory(processingDirectory);
+                return;
+            }
+
+            exeRename.MoveTo(Path.Combine(exeRename.Directory.ToString(), dataPath.Name.Replace("_Data", "") + ".exe"));
+
+
+            DirectoryInfo goodBuilddirectory = directory.CreateSubdirectory("GoodBuild");
+            DeleteContentsOfDirectory(goodBuilddirectory);
+            CopyFilesRecursively(processingDirectory.ToString(), goodBuilddirectory.ToString());
+            DeleteContentsOfDirectory(processingDirectory);
+            //deleteContentsOfDirectory(Stagingdirectory); TODO
+        }
+        catch (Exception e)
+        {
+            errors.Invoke(" an Error happened > " + e);
+            DeleteContentsOfDirectory(processingDirectory);
+            //deleteContentsOfDirectory(Stagingdirectory); TODO
+        }
+    }
+
+    public bool ScanFolder(DirectoryInfo @unsafe, DirectoryInfo saveFiles, Action<string> info, Action<string> errors)
+    {
+        var goodFiles = saveFiles.GetFiles().Select(x => x.Name).ToList();
+
+        info.Invoke("Provided files " + string.Join(",", goodFiles));
+
+        CopyFilesRecursively(saveFiles.ToString(), @unsafe.ToString());
+
+        var files = @unsafe.GetFiles();
+
+
+        List<string> multiAssemblyReference = new List<string>();
+
+        foreach (var file in files)
+        {
+            if (goodFiles.Contains(file.Name)) continue;
```

Comment:
Small nitpick, but I also agree with Codacy here.

Reply:
"Reformat the code to have only one statement per line "
imo Small stuff like continuing return is fine for in-line